pkg/shardddl/optimism/lock.go

```diff
@@ -180,8 +185,9 @@ func (l *Lock) TrySync(callerSource, callerSchema, callerTable string,
 		nextTable = schemacmp.Encode(newTI)
 		// special case: check whether DDLs making the schema become part of larger and another part of smaller.
 		if _, err = prevTable.Compare(nextTable); err != nil {
-			return emptyDDLs, terror.ErrShardDDLOptimismTrySyncFail.Delegate(
+			err = terror.ErrShardDDLOptimismTrySyncFail.Delegate(
```

Comment:
Do we need this? err is a returned variable in L95.

Reply:
address in dfab765